### PR TITLE
Add aarch64-apple-darwin target to CI yaml

### PR DESCRIPTION
**Pull request by fanninpm**
Alternate implementation of #762 that adds toolchains on GitHub's side. To test this locally, you need [act](https://github.com/nektos/act) and the ability to build your own Docker image with the correct tools. The recommended images are [either bulky or non-functional (or are unavailable from DockerHub)](https://github.com/catthehacker/docker_images/issues/13).

Closes #762

### CODE REVIEW

**Comment by JohnTitor** [comment]
Could you cherry-pick the commit from #754 so that we can make sure it works well?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ jobs:
         run: |
           rustup set profile minimal
           rustup toolchain install nightly
+          rustup target add --toolchain nightly aarch64-apple-darwin
```
Comment (JohnTitor) [inline .github/workflows/main.yml:16]: A comment describing why this is needed would be great.

Comment (fanninpm) [reply .github/workflows/main.yml:17]: Is d10e6af sufficient?

Comment (JohnTitor) [reply .github/workflows/main.yml:17]: Yup, thanks!

**Pull request merged**